src/shared/ui/image/image.tsx

```diff
@@ -2,20 +2,25 @@
 import { FC, useState } from 'react';
 import { IS_DEV } from './constants';
 import { DecodingAttr, FetchPriorityAttr, ImageProps, LoadingAttr } from './types';
+import checkForSuitable from './utils/checkForSuitable';
 import convertToWebp from './utils/convertToWebp';
 import generateSizes from './utils/generateSizes';
 import generateSrcSet from './utils/generateSrcSet';
 
 const Image: FC<ImageProps> = ({ alt, src = '', lazy = 'true', ...props }) => {
-  const srcWebp = convertToWebp(src);
-  const [srcSet, setSrcSet] = useState(() => (IS_DEV ? undefined : generateSrcSet(srcWebp)));
-  const [sizes, setSizes] = useState(() => (IS_DEV ? undefined : generateSizes()));
+  const isSuitable = checkForSuitable(src);
+
+  const srcAttr = isSuitable ? src : convertToWebp(src);
+
+  const [srcSet, setSrcSet] = useState(() =>
```

Comment:
Please, write it on 1 line

Reply:
To be honest, the changes served more to discuss approaches, sorry)